Automated simulation

JAndringa opened the pull request:
added bash scripts that automate the extremely tedious task of starting up all the different programs needed to run a simulation. also changed the file name from automatedDemo to automatedSimulation because it's not a demo :)

Florisheinen1 committed:
Update README

By removing outdated pieces of text

Florisheinen1 committed:
Update README

By removing outdated pieces of text

Florisheinen1 committed:
Move install commands to script

And removed the json dependency

Florisheinen1 committed:
Move install commands to script

And removed the json dependency

Florisheinen1 committed:
Remove unused CMake module

We did not use the ibx websockets anymore, so I removed the dependency

Florisheinen1 committed:
Remove unused CMake module

We did not use the ibx websockets anymore, so I removed the dependency

Florisheinen1 committed:
Clean up install script

Sorted dependencies a bit

Florisheinen1 committed:
Clean up install script

Sorted dependencies a bit

Florisheinen1 committed:
Fix protobuf to 3.19.x

Florisheinen1 committed:
Fix protobuf to 3.19.x

JAndringa committed:
added an automated simulation

added bash scripts that automate the extremely tedious task of starting up all the different programs needed to run a simulation. also changed the file name from automatedDemo to automatedSimulation because it's not a demo :)

silashoevers commented:
My first remark is how you are expecting files and folders to have a specific name, while the script can be made more rigid if you for example make use of regular expressions (e.g. ./ssl-game* instead of ssl-game-controller-[some version number]) to match the constant name of a file. A proposition for the future is to create an overarching repo that has suite as a submodule so you can enforce folder names

JAndringa committed:
improved the automated simulation script

made the automated simulation script more rigid by using a wildcard pattern

JAndringa committed:
deleted unused function

function that approaches trajectory by time was unused. also deleted all functions that were only used within this function

JAndringa committed:
deleted unused function (see previous commit)

function that approaches trajectory by time was unused. also deleted all functions that were only used within this function

JAndringa committed:
Revert "deleted unused function (see previous commit)"

This reverts commit ba9ea3921ec000164450709e3a5b3e31e3f0cad8.

JAndringa committed:
Revert "deleted unused function"

This reverts commit 5c0de6ea54243d8613d18b49ff53ef46bc1bc4f9.

JAndringa committed:
Merge branch 'development' into automated_simulation

JAndringa committed:
changed automatedSimulation file to bin and made build script scale to available cores

amount of cores used for building was hard coded to 9 cores, which poses a problem when you have less than 9 cores available, for example when using a quad-core processor. Fixed this by scaling it to the amaount of available cores. Also the name 'bin' is more intuitive than 'automatedSimulation', so I changed the name of this file. 'RunSimulation' should now be run from the roboteam_suite file instead, for example: '$ ./bin/RunSimulation.sh'.

silashoevers commented:
Everything seems to be spun up correctly now, though a problem of the AutoRef seemingly not getting vision persists. I will try to fix this.

Commit:
Fixed paths and removed duplicate runnables

Commit:
Updated RunRoboTeam script to conform to suggested file structure
